File: util/excel_to_json.js
const XLSX = require('xlsx');
const fs = require('fs');

function excelToJson(excelFile, outputFile) {
    try {
        // 엑셀 파일 읽기
        const workbook = XLSX.readFile(excelFile);
        
        // 결과를 저장할 객체
        const result = {};
        
        // 각 시트 처리
        workbook.SheetNames.forEach(sheetName => {
            const worksheet = workbook.Sheets[sheetName];
            
            // 시트를 JSON으로 변환
            const jsonData = XLSX.utils.sheet_to_json(worksheet, { 
                header: 1,
                defval: null 
            });
            
            if (jsonData.length > 0) {
                const headers = jsonData[0];
                const data = jsonData.slice(1);
                
                result[sheetName] = {
                    columns: headers,
                    data: data
                };
            }
        });
        
        // JSON 파일로 저장
        fs.writeFileSync(outputFile, JSON.stringify(result, null, 2), 'utf8');
        
        console.log(`변환 완료! ${excelFile} -> ${outputFile}`);
        console.log(`시트 개수: ${Object.keys(result).length}`);
        
        Object.keys(result).forEach(sheetName => {
            const sheet = result[sheetName];
            console.log(`  - ${sheetName}: ${sheet.data.length}행, ${sheet.columns.length}열`);
        });
        
    } catch (error) {
        console.error(`오류 발생: ${error.message}`);
    }
}

// 실행
const excelFile = '황금경로.xlsx';
const outputFile = '황금경로.json';

if (fs.existsSync(excelFile)) {
    excelToJson(excelFile, outputFile);
} else {
    console.log(`파일을 찾을 수 없습니다: ${excelFile}`);
} 

File: util/excel_to_json.py
import pandas as pd
import json
import os

def excel_to_json(excel_file, output_file):
    """
    엑셀 파일을 JSON으로 변환하는 함수
    """
    try:
        # 엑셀 파일의 모든 시트 읽기
        excel_data = pd.read_excel(excel_file, sheet_name=None)
        
        # 결과를 저장할 딕셔너리
        result = {}
        
        for sheet_name, df in excel_data.items():
            # DataFrame을 딕셔너리로 변환
            # NaN 값을 None으로 변환
            df_clean = df.where(pd.notnull(df), None)
            
            # 시트별로 데이터 저장
            result[sheet_name] = {
                'columns': df_clean.columns.tolist(),
                'data': df_clean.values.tolist()
            }
        
        # JSON 파일로 저장
        with open(output_file, 'w', encoding='utf-8') as f:
            json.dump(result, f, ensure_ascii=False, indent=2)
        
        print(f"변환 완료! {excel_file} -> {output_file}")
        print(f"시트 개수: {len(result)}")
        for sheet_name in result.keys():
            print(f"  - {sheet_name}: {len(result[sheet_name]['data'])}행, {len(result[sheet_name]['columns'])}열")
            
    except Exception as e:
        print(f"오류 발생: {e}")

if __name__ == "__main__":
    excel_file = "황금경로.xlsx"
    output_file = "황금경로.json"
    
    if os.path.exists(excel_file):
        excel_to_json(excel_file, output_file)
    else:
        print(f"파일을 찾을 수 없습니다: {excel_file}") 